src/main/java/hudson/plugins/pmd/PmdPublisher.java

```diff
@@ -25,14 +24,20 @@
  * @author Ulli Hafner
  */
 public class PmdPublisher extends HealthAwarePublisher {
+
     private static final long serialVersionUID = 6711252664481150129L;
```

Comment:
If you look carefully in some of the well written long evolved plugins you will see the "fun" "dancing games" that happen in `readResolve()` methods and you will also see some `/*effectively final*/` comments where the field would be regarded as `final` but because it needs to be populated from the `readResolve()` method it isn't... an alternative method would be to have the `readResolve()` return a replacement instance... but beware class evolution as it is very easy to forget to update the copy constructor used in the `readResolve()` when you add a new field... so sometimes it is safer to have the field effectively final (i.e. only written from either constructors or `readResolve()`)